awx/main/management/commands/host_metrics.py

```diff
@@ -0,0 +1,69 @@
+from django.core.management.base import BaseCommand
+from dateutil import parser
+from django.utils.timezone import now
+import pytz
+from datetime import datetime, timezone
+from awx.main.models.inventory import HostMetrics
+import json
+
+
+class Command(BaseCommand):
+
+    help = 'This is for offline licensing usage'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--since', dest='since', action='store', help='Start date for data')
+        parser.add_argument('--until', dest='until', action='store', help='End date for data')
+        parser.add_argument('--json', action='store_true', help='Select output as JSON')
+
+    def handle(self, *args, **options):
+        opt_since = options.get('since')
+        opt_until = options.get('until')
+
+        # user gives no arguments
+        if opt_since is None and opt_until is None:
+            print("No Arguments received")
+            return None
+
+        since = None
+        if opt_since:
+            since = parser.parse(opt_since)
+            if since.tzinfo is None:
+                since = since.replace(tzinfo=timezone.utc)
+
+        until = None
+        if opt_until:
+            until = parser.parse(opt_until)
+            if until.tzinfo is None:
+                until = until.replace(tzinfo=timezone.utc)
+
+        if (since is None) and (until is not None):
+            filter_kwargs = {"last_automation__lte": until}
+        elif (since is not None) and (until is None):
+            filter_kwargs = {"last_automation__gte": since}
+        else:
+            filter_kwargs = {"last_automation__gte": since, "last_automation__lte": until}
```

Comment:
Maybe this checking of cases could simplified by just having a set of conditions and adding them individually:

```python
filter_kwargs = {}
if since is not None:
    filter_kwargs['last_automation__gte'] = since
if until is not None:
    filter_kwargs['last_automation__lte'] = until
```

Reply:
I have made the necessary changes.